app/src/main/java/org/mozilla/fenix/tabhistory/TabHistoryViewHolder.kt

```diff
@@ -7,28 +7,34 @@ package org.mozilla.fenix.tabhistory
 import android.view.View
 import androidx.core.text.bold
 import androidx.core.text.buildSpannedString
-import androidx.core.view.isVisible
-import androidx.recyclerview.widget.RecyclerView
-import kotlinx.android.synthetic.main.history_list_item.view.*
+import kotlinx.android.synthetic.main.history_list_item.*
+import org.mozilla.fenix.library.LibrarySiteItemView
+import org.mozilla.fenix.utils.view.ViewHolder
 
 class TabHistoryViewHolder(
-    private val view: View,
+    view: View,
     private val interactor: TabHistoryViewInteractor
-) : RecyclerView.ViewHolder(view) {
+) : ViewHolder(view) {
+
+    private lateinit var item: TabHistoryItem
+
+    init {
+        itemView.setOnClickListener { interactor.goToHistoryItem(item) }
```

Comment:
This needs to be moved back to the bind method. Otherwise we're attempting to go to an item that isn't initialized yet

Reply:
Also it seems like we may need to set the listener on `history_layout` because this isn't working at all for me when its set on `itemView` for some reason. So it becomes `history_layout.setOnClickListener { interactor.goToHistoryItem(item) }` in `bind`

Reply:
This pattern is a bit better for perf because each listener is a new object allocation. If you're worried about `bind` being called after clicking, we can change `item` to be nullable instead of lateinit.

Reply:
Also agree on moving to history_layout, that's a typo :)

Reply:
> This pattern is a bit better for perf because each listener is a new object allocation. If you're worried about bind being called after clicking, we can change item to be nullable instead of lateinit.

Double check myself and you're right. :) We can leave it where it is.